internal/core/metadata/interfaces/db.go

```diff
@@ -97,13 +97,13 @@ type DBClient interface {
 	DeleteDeviceServiceById(id string) error
 
 	// Provision watcher
-	GetProvisionWatcherById(pw *contract.ProvisionWatcher, id string) error
-	GetAllProvisionWatchers(pw *[]contract.ProvisionWatcher) error
-	GetProvisionWatcherByName(pw *contract.ProvisionWatcher, n string) error
-	GetProvisionWatchersByProfileId(pw *[]contract.ProvisionWatcher, id string) error
-	GetProvisionWatchersByServiceId(pw *[]contract.ProvisionWatcher, id string) error
-	GetProvisionWatchersByIdentifier(pw *[]contract.ProvisionWatcher, k string, v string) error
-	AddProvisionWatcher(pw *contract.ProvisionWatcher) error
+	GetProvisionWatcherById(id string) (pw contract.ProvisionWatcher, err error)
```

Comment:
Convention thus far has been to not specify named return parameters in this interface.  Consistency issue -- calling out but don't consider this merge-blocking.  My preference would be to have named parameters for all the things.